lessons/advanced/otp-concurrency.md

```diff
@@ -149,79 +147,120 @@ iex> SimpleQueue.queue
+  end
+```
+
+#### Scheduling and Performing Job
+Next we set up a way to schedule our work and handle the call with [`handle_info/2`](https://hexdocs.pm/elixir/GenServer.html#c:handle_info/2):
 
 ```elixir
-defmodule LoggerHandler do
-  use GenEvent
+  def handle_info(:perform_job, {module, function, args, interval}) do
+    apply(module, function, args)
+    schedule_work(interval)
+    {:noreply, {module, function, args, interval}}
+  end
 
-  def handle_event({:msg, msg}, messages) do
-    IO.puts "Logging new message: #{msg}"
-    {:ok, [msg|messages]}
+  defp schedule_work(interval) do
+    Process.send_after(self(), :perform_job, interval)
   end
+```
+
+#### Running The Job Indefinitely
+And finally, we set up our `init/1` to just call those with the given interval:
 
-  def handle_call(:messages, messages) do
-    {:ok, Enum.reverse(messages), messages}
+```elixir
+  def init({module, function, args, interval}) do
+    schedule_work(interval)
+    {:ok, {module, function, args, interval}}
   end
-end
 ```
 
-### Using GenEvents
+Now, lets create a worker to pass into this worker for once we get it into our tree in `lib/silly_worker.ex`.
 
-With our handlers ready to go we need to familiarize ourselves with a few of GenEvent's functions.  The three most important functions are: `add_handler/3`, `notify/2`, and `call/4`.  These allow us to add handlers, broadcast new messages, and call specific handler functions respectively.
+#### Making A Worker Task
+This is very simple, we aren't going to be doing any real work but we can see the functionality.
+
+```elixir
+defmodule SillyWorker do
+  def do_work(job_name) do
+    # do some work
+    # ...
+    require Logger
+    Logger.log(:info, "#{job_name} Complete!")
+    :ok
+  end
+end
+```
 
-If we put it all together we can see our handlers in action:
+#### Setting Up Our Worker
+Finally we just set up our worker to run in the given interval we choose in our `application.ex` file:
 
 ```elixir
-iex> {:ok, pid} = GenEvent.start_link([])
-iex> GenEvent.add_handler(pid, LoggerHandler, [])
-iex> GenEvent.add_handler(pid, PersistenceHandler, [])
+    children = [
+      worker(SillyWorker.CronJob, [SillyWorker, :do_work, ["Job Number 1"], 1000]),
+    ]
+```
 
-iex> GenEvent.notify(pid, {:msg, "Hello World"})
-Logging new message: Hello World
-Persisting log message: Hello World
+Now if we just run IEx the worker will run that function on our module every second in its own process under the supervision tree:
 
-iex> GenEvent.call(pid, LoggerHandler, :messages)
-["Hello World"]
 ```
+$ iex -S mix
+iex(1)>
+20:39:27.732 [info]  Job Done
 
-See the official [GenEvent](http://elixir-lang.org/docs/stable/elixir/GenEvent.html#content) documentation for a complete list of callbacks and GenEvent functionality.
+20:39:28.232 [info]  Job Done
+```
+
+Thats it!
+Check out a working example of the CronJob app [here](https://github.com/ybur-yug/genserver_periodic_worker_example).
```

Comment:
We have refrained from linking to any individual's repos and I'd like to continue to do that.